### PR TITLE
adding counting methods

### PR DESCRIPTION
**Pull request by maryesima**

### CODE REVIEW

File: src/document/BasicDocument.java [hunk 1]
```diff
@@ -52,11 +53,12 @@ public int getNumWords()
...
-	public int getNumSentences()
-	{
-	    //TODO: Implement this method.  See the Module 2 support videos 
-        // if you need help.
-        return 0;
+	public int getNumSentences() {
+	    if (getText().isEmpty()) {
+	        return 0;
+        }
+	    String[] text = getText().split("[.!?]+");
```
Comment (etwei-linkedin) [inline src/document/BasicDocument.java:60]: format

Comment (etwei-linkedin) [inline src/document/BasicDocument.java:60]: The format of these 2 lines looks weird...

File: src/document/BasicDocument.java [hunk 2]
```diff
@@ -32,11 +32,12 @@ public BasicDocument(String text)
...
-	{
-		//TODO: Implement this method in week 2 according to the comments above.  
-		// See the Module 2 support videos if you need help.
-	    return 0;
+	public int getNumWords() {
+		if (getText().isEmpty()) {
+			return 0;
+		}
+		String[] text = getText().split("[.!?, '^0-9]+");
+		return text.length;
```
Comment (etwei-linkedin) [inline src/document/BasicDocument.java:40]: looks like these are 8 spaces tab?

File: src/document/Document.java [hunk 3]
```diff
@@ -66,8 +65,25 @@ protected int countSyllables(String word)
...
 		// TODO: Implement this method so that you can call it from the 
 	    // getNumSyllables method in BasicDocument (module 2) and 
-	    // EfficientDocument (module 3).
-	    return 0;
+	    // EfficientDocument (module 3)
+		int numSyllables = 0;
+		boolean isPrevCharVowel = false;
+		Set<String> vowels = new HashSet<>(Arrays.asList("a", "e", "i", "o", "u", "y"));
+		for (int i = 0; i < word.length(); i++) {
+			String c = word.toLowerCase().substring(i, i+1);
```
Comment (etwei-linkedin) [inline src/document/Document.java]: I would go with String#charAt method. Then later when you need to compare equality from a char to a char, you can easily do char1 == char2.

File: src/document/Document.java [hunk 4]
```diff
@@ -66,8 +65,25 @@ protected int countSyllables(String word)
 	{
 		// TODO: Implement this method so that you can call it from the 
 	    // getNumSyllables method in BasicDocument (module 2) and 
-	    // EfficientDocument (module 3).
-	    return 0;
+	    // EfficientDocument (module 3)
+		int numSyllables = 0;
+		boolean isPrevCharVowel = false;
+		Set<String> vowels = new HashSet<>(Arrays.asList("a", "e", "i", "o", "u", "y"));
```
Comment (etwei-linkedin) [inline src/document/Document.java]: Maybe a `Set<Character>`?

File: src/document/Document.java [hunk 5]
```diff
@@ -66,8 +65,25 @@ protected int countSyllables(String word)
...
-	    // EfficientDocument (module 3).
-	    return 0;
+	    // EfficientDocument (module 3)
+		int numSyllables = 0;
+		boolean isPrevCharVowel = false;
+		Set<String> vowels = new HashSet<>(Arrays.asList("a", "e", "i", "o", "u", "y"));
+		for (int i = 0; i < word.length(); i++) {
+			String c = word.toLowerCase().substring(i, i+1);
+			if (vowels.contains(c)) {
+				if (!isPrevCharVowel) {
```
Comment (etwei-linkedin) [inline src/document/Document.java:75]: Can you put your algorithms in the method description?

**Review by etwei-linkedin — COMMENTED** [review]
Looks like your editor is using some default formtter. Highly recommend going with [Google java formatter](https://plugins.jetbrains.com/plugin/8527-google-java-format) which you can download from your intellij marketplace for free.

**Comment by maryesima** [comment]
I've updated my files with the google formatter in the second PR so that I don't have merge conflicts in this one, is that ok? Am I good to merge this one in that case? Thanks!

**Comment by maryesima** [comment]
created a new PR to add the google formatter changes

**Pull request merged**